GoogleForCreators/web-stories-wp
# Animation Panel: Effect dropdown a11y

## Summary
adds initial keyboard functionality to animation dropdown. 

From the ticket: 
> all the same keyboard-functionality as for the “regular” advanced dropdown (space, enter or arrow-down to open dropdown, auto-focus first item when opening, arrow keys to navigate in dropdown, enter/space to select option, esc to close dropdown, and auto-focus dropdown button itself on close).

## Relevant Technical Choices
- Since this dropdown's options are hardcoded I added some references to use as maps to check for index location to preserve existing selection when the menu becomes open and allow for keyboard arrow functionality to match other dropdowns. 
- The only place that the form/dropDown is getting used is in the animation panel for this dropdown, so I fixed the lack of outline at the source because it doesn't have any other use cases. 


## To-do

- [x] Override tab functionality when within the dropdown but allow escape
- [x] limiting background animation list to active options since those are disabled according to content.

## User-facing changes
- On open of effect dropdown menu, if no option is selected yet the first option will be focused 
- On open of effect dropdown menu, if you have an existing option selected from earlier it will be in focus 
- You can use up and down arrows to navigate the menu, once you reach the last item (or the first) you cannot arrow further, you stay in the menu 
- If you press 'esc' in the dropdown menu, the menu will close and the animation panel for that element will stay open. 
- Focused option will shift with keyboard or with cursor. 

## Testing Instructions
- Open the editor and add an element, open the animation panel and use the effect dropdown with a keyboard and also with a mouse, see that it works as you would expect. 

---



Addresses #5524
